drivers/display/display_renesas_ra.c

```diff
@@ -0,0 +1,417 @@
+			(void)memcpy(dst, src, desc->width * data->pixel_size);
+			dst += (config->width * data->pixel_size);
+			src += (desc->pitch * data->pixel_size);
+		}
+	}
+
+	if (data->front_buf == l_pend_buf) {
+		return 0;
+	}
+
+	k_sem_reset(&data->sem);
+
+	data->pend_buf = l_pend_buf;
+
+	err = R_GLCDC_BufferChange(&data->display_ctrl, (uint8_t *)data->pend_buf,
+				   DISPLAY_FRAME_LAYER_1);
+	if (err) {
+		LOG_ERR("GLCDC buffer change failed");
+		return -EIO;
+	}
+
+	k_sem_take(&data->sem, K_FOREVER);
+
+	return 0;
+}
+
+static int ra_display_read(const struct device *dev, const uint16_t x, const uint16_t y,
+			   const struct display_buffer_descriptor *desc, void *buf)
+{
+	struct display_ra_data *data = dev->data;
+	const struct display_ra_config *config = dev->config;
+	uint8_t *dst = buf;
+	const uint8_t *src = data->front_buf;
+	uint16_t row;
+
+	/* src = pointer to upper left pixel of the rectangle to be read from frame buffer */
+	src += (x * data->pixel_size);
+	src += (y * config->width * data->pixel_size);
+
+	for (row = 0; row < desc->height; row++) {
+		(void)memcpy(dst, src, desc->width * data->pixel_size);
+		src += (config->width * data->pixel_size);
+		dst += (desc->pitch * data->pixel_size);
+	}
+
+	return 0;
+}
+
+static int ra_display_blanking_on(const struct device *dev)
+{
+	const struct display_ra_config *config = dev->config;
+	int err;
+
+	if (config->backlight_gpio.port != NULL) {
+		err = gpio_pin_set_dt(&config->backlight_gpio, 0);
+		if (err) {
+			LOG_ERR("Disable backlight failed! (%d)", err);
+			return err;
+		}
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int ra_display_blanking_off(const struct device *dev)
+{
+	const struct display_ra_config *config = dev->config;
+	int err;
+
+	if (config->backlight_gpio.port != NULL) {
+		err = gpio_pin_set_dt(&config->backlight_gpio, 1);
+		if (err) {
+			LOG_ERR("Enable backlight failed! (%d)", err);
+			return err;
+		}
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static void ra_display_get_capabilities(const struct device *dev,
+					struct display_capabilities *capabilities)
+{
+	const struct display_ra_config *config = dev->config;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = config->width;
+	capabilities->y_resolution = config->height;
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+	capabilities->supported_pixel_formats =
+		PIXEL_FORMAT_RGB_888 | PIXEL_FORMAT_ARGB_8888 | PIXEL_FORMAT_RGB_565;
+	capabilities->current_pixel_format = (config->pixel_format == PIXEL_FORMAT_RGB_888)
+						     ? PIXEL_FORMAT_ARGB_8888
+						     : config->pixel_format;
+}
+
+static int ra_display_set_pixel_format(const struct device *dev,
+				       const enum display_pixel_format pixel_format)
+{
+	const struct display_ra_config *config = dev->config;
+
+	if (pixel_format == config->pixel_format) {
+		return 0;
+	}
+	LOG_WRN("Pixel format changes must be set in dts at build time.");
```

Comment:
I think this is not an issue. This warning is for the display controller and with display_ili9806e_dsi for the shield driver. So, it can be understood that:

- This driver should use build-time config on dts to synchronous with the display_ili9806e_dsi config. 
- And display_ili9806e_dsi pixel_format cannot supported (because it uses output format of display controller)

Reply:
This should be a LOG_ERR, not warning IMO

Reply:
Done. I updated it to LOG_ERR.